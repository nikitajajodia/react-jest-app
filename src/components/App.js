import React from 'react';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      gifts: [],
    };
  }

  addGift = () => {
    const {
      gifts,
    } = this.state;

    const ids = gifts.map(gift => gift.id);
    const maxId = ids.length > 0 ? Math.max(...ids) : 0;

    gifts.push({
      id: maxId + 1,
    });

    this.setState({
      gifts,
    });
  }

  render() {
    const {
      gifts,
    } = this.state;
    return (
      <div>
        <h2>
          Gift Giver
        </h2>
        <div className="gift-list">
          {
            gifts.map(gift => (
              <div key={gift.id}>
                {gift.id}
                test
              </div>
            ))
          }
        </div>
        <button className="btn-add" onClick={this.addGift} type="button">
          Add Gift
        </button>
      </div>
    );
  }
}

export default App;
