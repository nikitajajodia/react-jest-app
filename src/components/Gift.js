import React from 'react';
import PropTypes from 'prop-types';

class Gift extends React.Component {
  constructor() {
    super();
    this.state = {
      person: '',
      present: '',
    };
  }

  render() {
    const {
      person, present,
    } = this.state;
    const {
      removeGift,
      gift,
    } = this.props;
    console.log(person, present);
    return (
      <div>
        <form>
          Person
          <input
            type="text"
            className="input-person"
            onChange={event => this.setState({ person: event.target.value })}
          />
          Present
          <input
            type="text"
            className="input-present"
            onChange={event => this.setState({ present: event.target.value })}
          />
        </form>
        <button
          onClick={() => removeGift(gift.id)}
          type="button"
          className="remove-gift"
        >
            Remove Gift
        </button>
      </div>
    );
  }
}

export default Gift;

Gift.propTypes = {
  removeGift: PropTypes.func.isRequired,
  gift: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};
