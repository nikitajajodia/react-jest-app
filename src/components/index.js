import Header from './Header';
import Footer from './Footer';
import App from './App';
import Gift from './Gift';

export {
  Header,
  Footer,
  App,
  Gift,
};
