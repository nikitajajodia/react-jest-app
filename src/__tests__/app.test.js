import React from 'react';

import { App } from '../components';

describe('App', () => {
  const app = shallow(<App />);

  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('initialises the state with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
  });

  describe('when clicking on add gift button', () => {
    // is executed before each test running in this block
    beforeEach(() => {
      app.find('.btn-add').simulate('click');
    });

    // is executed after each test running in this block
    afterEach(() => {
      app.setState({ gifts: [] });
    });

    it('adds a new gift to state', () => {
      expect(app.state().gifts).toEqual([{ id: 1 }]);
    });

    it('addas a new gift to the rendered list', () => {
      expect(app.find('.gift-list').children().length).toEqual(1);
    });
  });
});
