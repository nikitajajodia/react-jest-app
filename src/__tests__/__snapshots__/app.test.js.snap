// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    Gift Giver
  </h2>
  <div
    className="gift-list"
  />
  <button
    className="btn-add"
    onClick={[Function]}
    type="button"
  >
    Add Gift
  </button>
</div>
`;
